### PR TITLE
Added Bootstrap options for dashboard ssl key and cert

### PR DESCRIPTION
**Pull request by sunilangadi2**
Signed-off-by: Sunil Angadi <sangadi@redhat.com>

# Description
Bootstrap with dashboard options
- dashboard-key
- dashboard-crt
- dashboard-password-noupdate
Success logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1625217379701
Polarion link: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574421

### CODE REVIEW

**Comment by sunilangadi2** [comment]
Passing these options **dashboard-key** and **dashboard-crt** in bootstrap 
Internally it calls following command to set ssl provided cert and key for the dashboard security authority
$ ceph dashboard set-ssl-certificate -i dashboard.crt
$ ceph dashboard set-ssl-certificate-key -i dashboard.key
Some of the validation part is already available in `validate_dashboard` method in `test_bootstrap.py`
actually it needs checking certification comparison from the dashboard GUI itself 
so it's quite difficult to validate that from the code hence no validation tests were added.

**Review by sunilkumarn417 — COMMENTED** [review]
Hey @sunilangadi2 ,

Code changes looks good to me.
but for generic approach, I think we can tweak few code changes as per the comments.

File: ceph/ceph_admin/bootstrap.py [hunk 1]
```diff
@@ -58,6 +58,30 @@ def construct_registry(cls, registry: str, json_file: bool = False):
...
+        sudo=True,
+        cmd="yum install -y openssl",
+    )
+
+    # Generating key and cert using openssl in /home/cephuser
+    cls.installer.exec_command(
+        sudo=True,
+        cmd='openssl req -new -nodes -x509 \
+            -subj "/O=IT/CN=ceph-mgr-dashboard" -days 3650 \
+            -keyout /home/cephuser/dashboard.key \
```
[the remark this answers is not in the document]
Comment (sunilangadi2) [reply ceph/ceph_admin/bootstrap.py:80]: Ok sure

File: ceph/ceph_admin/bootstrap.py [hunk 2]
```diff
@@ -58,6 +58,30 @@ def construct_registry(cls, registry: str, json_file: bool = False):
     return config_dict_to_string(reg_args)
 
 
+def construct_key_cert(cls):
+    """
+    Construct dashboard key and certificate files for bootstrapping cluster
```
[the remark this answers is not in the document]
Comment (sunilangadi2) [reply ceph/ceph_admin/bootstrap.py:63]: ok sure

File: ceph/ceph_admin/bootstrap.py [hunk 3]
```diff
@@ -167,6 +191,10 @@ def bootstrap(self: CephAdmProtocol, config: Dict):
         if registry_json:
             cmd += construct_registry(self, registry_json, json_file=True)
 
+        # Construct dashboard cert and key if bootstrap cli have this options
+        if all(key in args for key in ("dashboard-key", "dashboard-crt")):
```
[the remark this answers is not in the document]
Comment (sunilangadi2) [reply ceph/ceph_admin/bootstrap.py:195]: Yes these are mutually needed in order to access certificate you need there keys

File: suites/pacific/cephadm/tier1_ssl_dashboard_port.yaml [hunk 4]
```diff
@@ -22,6 +22,9 @@
 #       - fsid : f64f341c-655d-11eb-8778-fa163e914bcc
 #       - initial-dashboard-user: admin123
 #       - initial-dashboard-password: admin@123,
+#       - dashboard-password-noupdate
```
[the remark this answers is not in the document]
Comment (sunilangadi2) [reply suites/pacific/cephadm/tier1_ssl_dashboard_port.yaml:25]: Yes

**Comment by sunilangadi2** [comment]
@sunilkumarn417 Thanks for the review, Updated PR with your suggestion
Success Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1625567463730

**Review by sunilkumarn417 — APPROVED** [review]
@sunilangadi2 Thanks for working on the review comments.

Code changes looks good to me.

**Pull request merged**